data-service/src/main/java/com/amazonaws/blox/dataservice/repository/model/EnvironmentTargetVersionDDBRecord.java

```diff
@@ -46,4 +61,24 @@
   private String cluster;
 
   @DynamoDBVersionAttribute private Long recordVersion;
+
+  private String environmentVersion;
+
+  @DynamoDBTypeConverted(converter = InstantDDBConverter.class)
```

Comment:
What's the benefit we're getting from duplicating the other fields of EnvironmentDDBRecord in EnvironmentTargetVersionDDBRecord? Since the individual versioned environments are immutable, we could do a two-step lookup for environmentVersion from this table, and the actual values from the other table.

That would save us from having to keep the two data models in sync, and potentially allow us to keep pointers to other useful versions in this table for other uses (e.g. "previousVersion" for rollbacks without an explicit version, or "lastCreatedVersion" for keeping track of the newest update to the environment in the absence of ordered version numbers.

Reply:
The advantage is that we don't do two reads. We can always keep extra information here too but I'm open to doing it either way.

Reply:
As mentioned in the other comment, I think if you want to avoid reading the values provided by the user to createEnvironmentTargetVersion, you'd have to do the read to get the values anyway...

Reply:
you would in the case of create but not describe if we have them in both places